src/com/pentalog/pentastagiu/homework2/HomeworkApp.java

```diff
@@ -1,7 +1,113 @@
 package com.pentalog.pentastagiu.homework2;
 
+import com.pentalog.pentastagiu.homework2.bank.BankAccount;
+import com.pentalog.pentastagiu.homework2.bank.NotEnoughMoneyException;
+import com.pentalog.pentastagiu.homework2.chat.Board;
+import com.pentalog.pentastagiu.homework2.chat.User;
+import com.pentalog.pentastagiu.homework2.shop.Candy;
+import com.pentalog.pentastagiu.homework2.shop.Customer;
+import com.pentalog.pentastagiu.homework2.shop.Book;
+import com.pentalog.pentastagiu.homework2.shop.Membership;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
 public class HomeworkApp {
+
+    public static void testBankAccountApp() {
+
+        // 1st - test constructor
+        try {
+            BankAccount invalidBankAccount = new BankAccount(4);
+        } catch (NotEnoughMoneyException e) {
+            System.out.println(e);
+        }
+        try {
+            BankAccount bankAccount = new BankAccount(15);
+            BufferedReader readCommand = new BufferedReader(new InputStreamReader(System.in));
+            String command = "start";
+            while (!"0".equals(command)) {
+
+                switch (command) {
+                    case "0":
+                        command = "0";
+                        break;
+                    case "1":
+                        bankAccount.displaySold();
+                        break;
+                    case "2":
+                        System.out.print("Introduce the value you want to withdraw: ");
+                        try {
+                            double value = Double.parseDouble(readCommand.readLine());
+                            bankAccount.withdraw(value);
+                        } catch (NumberFormatException e) {
+                            System.out.println("Enter a numeric value!");
+                        }
+                        break;
+                    case "3":
+                        System.out.print("Introduce the value you want to deposit: ");
+                        try {
+                            double value = Double.parseDouble(readCommand.readLine());
+                            bankAccount.deposit(value);
+                        } catch (NumberFormatException e) {
+                            System.out.println("Enter a numeric value!");
+                        }
+                        break;
+                    default:
```

Comment:
e o practica buna sa ai ramura de default, dar in cazul asta, cand nu implementezi nimic pentru cazul default si toate celelalte alternative decat cele tratate in case-uri pot fi ignorate, nu e nevoie sa o pui; 
de obicei se pune cand pe case-uri tratezi diverse cazuri, iar pe default vrei sa prinzi o valoare neasteptata si sa adaugi un throw new InvalidOptionException("The option selected is invalid.")